deeplake/enterprise/libdeeplake_query.py

```diff
@@ -35,10 +36,20 @@ def query(dataset, query_string: str):
         >>> ds_train = deeplake.load('hub://activeloop/coco-train')
         >>> query_ds_train = query(ds_train, "(select * where contains(categories, 'car') limit 1000) union (select * where contains(categories, 'motorcycle') limit 1000)")
     """
-    ds = dataset_to_libdeeplake(dataset)
+    if isinstance(dataset, DeepLakeQueryDataset):
+        ds = dataset.indra_ds
+    else:
+        ds = dataset_to_libdeeplake(dataset)
     dsv = ds.query(query_string)
-    indexes = dsv.indexes
-    return dataset[indexes]
+    try:
+        indexes = dsv.indexes
+        return dataset[indexes]
+    except RuntimeError:
```

Comment:
why is this produced?

Reply:
When getting `indexes` on non-linear view, indra raises an exception.